waku/v2/api/filter.go

```diff
@@ -229,5 +158,12 @@ func (apiSub *Sub) multiplex(subs []*subscription.SubscriptionDetails) {
 				apiSub.DataCh <- env
 			}
 		}(subDetails)
+		go func(subDetails *subscription.SubscriptionDetails) {
+			_, ok := <-subDetails.Closing
```

Comment:
Since apiSub wraps more than 1 subscription(i.e status-go can ask to subscribe for a filter for 2 peers) which would result in 2 subscriptions being created and wrapper in apiSub. 
Since Closing channel is used to indicate when a subscription is about to be closed, this has to be multiplexed into a single channel in order for apiSub to take action (i.e resubscribe).

There may be a better naming convention for this to make it more clear.